### PR TITLE
RPC:recovery

### PR DESCRIPTION
**Pull request by kewde**

### CODE REVIEW

**Comment by rynomster** [comment]
Please can you resolve the commit errors. It seems you and Pierre have some work that's overlapping, I'd suggest merging master into the recovery branch, and fixing the errors. I started, but now its conflicting again.

**Comment by kewde** [comment]
Going to rebase this branch on master.

**Comment by kewde** [comment]
I'm going to implement the functionality of the firsttime modal in here too

File: src/app/modals/recoverwallet/recoverwallet.component.html [hunk 1]
```diff
@@ -6,13 +6,104 @@
   </div>
 </div>
 
+<hr style="margin-top: 100px;">
```
Comment (rynomster) [inline src/app/modals/recoverwallet/recoverwallet.component.html]: Please don't use inline styles. Its difficult to work with.

File: src/app/modals/recoverwallet/recoverwallet.component.html [hunk 2]
```diff
@@ -6,13 +6,104 @@
...
+    </div>
+    <div class="col-md-2">
+      <input [(ngModel)]="words[23]" type="text" name="recovery" [disabled]="isDisabled" class="form-control recovery-word">
+    </div>
+  </div>
+</div>
+
+<hr style="margin-top: 40px;">
+
+<!--
```
Comment (rynomster) [inline src/app/modals/recoverwallet/recoverwallet.component.html]: Why are we not making use of the app-passphrase component?

File: src/app/modals/recoverwallet/recoverwallet.component.spec.ts [hunk 3]
```diff
@@ -3,13 +3,23 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { RecoverwalletComponent } from './recoverwallet.component';
 import { ModalsModule } from '../modals.module';
 
+import { ElectronService } from 'ngx-electron';
```
Comment (rynomster) [inline src/app/modals/recoverwallet/recoverwallet.component.spec.ts]: Use RpcModule instead.

File: src/app/modals/recoverwallet/recoverwallet.component.html (hunk 1 above)
Comment (kewde) [inline src/app/modals/recoverwallet/recoverwallet.component.html]: Will fix, I ported that over from master tho

File: src/app/modals/recoverwallet/recoverwallet.component.html (hunk 2 above)
Comment (kewde) [inline src/app/modals/recoverwallet/recoverwallet.component.html]: I purposefully left it out for the time being until I got to work on the firsttime dialog, where it's re-used.. I'm now creating that, which will also be used here.

**Comment by rynomster** [comment]
I've added some lint fixes, but there is an error regarding splitAndFill, and the code review has been ignored

**Comment by rynomster** [comment]
it seems ngx-tooltip is not maintained. will use bootstrap tooltips

**Comment by rynomster** [comment]
the RPC import works, but the user experience is terrible. We definitely need to work on this.

**Pull request merged**